020-Connect/020-JavaScript/020-prisma.mdx

```diff
@@ -0,0 +1,74 @@
+---
+title: Prisma with Xata
+navTitle: Prisma
+keywords: ['orm', 'prisma', 'typescript', 'query builder']
+description: Use the Prisma integration as a TypeScript ORM
+slug: integrations/prisma
+published: true
+status: beta
+---
+
+[Prisma ORM](https://www.prisma.io/orm) is a TypeScript ORM that is focused on type safety and developer experience.
+
+### Installation
+
+Install prisma client. You can use the following package managers:
+
+<TabbedCode tabs={['npm', 'pnpm', 'yarn']}>
+```sh
+npm install prisma
+```
+
+```sh
+pnpm add prisma
+```
+
+```sh
+yarn add prisma
+```
+
+</TabbedCode>
+
+### Usage
+
+You can use Prisma with Xata with the default postgres client. To do so, you need to define the schema in the `schema.prisma` file. Here is an example of a schema:
+
+```prisma
+datasource db {
+  provider = "postgresql"
+  url      = env("DATABASE_URL")
+}
+```
+
+You can manually define the schema or use the introspection feature to generate the schema from the database.
+
+To introspect the database, you can use the following command:
+
+```sh
+prisma db pull
+```
+
+To generate the client, you can use the following command:
+
+```sh
+prisma generate
+```
+
+After generating the client, you can use it in your code. Here is an example of using the client:
+
+```ts
+import { PrismaClient } from '@prisma/client';
+
+const prisma = new PrismaClient({});
+
+const result = await prisma.users.findMany({
+  where: {
+    name: 'Alice'
+  }
+});
+```
+
+### Limitations
+
+- Xata File Storage is read-only and only shows file metadata. You need to use the SDK to upload and download files.
+- Prisma migrate is not fully supported. You need to use the Xata CLI to manage the database migrations.
```

Comment:
Added a link to https://xata.io/docs/migrations . Those docs are not fully fleshed out, we probably need to enhance them as well a bit. Also added "yet" for our intentions here :) 

```suggestion
- Prisma migrate is not yet fully supported. You need to use the Xata CLI to manage the [database migrations](/docs/migrations).
```